### DISCUSSION TITLE
Need some more documentation

### DISCUSSION THREAD

**Discussion by ensemblebd** [maxtoroq/DbExtensions#45] [Q&A] [answered]
I mean don't get me wrong, it's great for what it does cover.

1) How the heck does one perform an insert statement? 
Yes there's the documented **SQL.INSERT_INTO** method, however that doesn't have a connection string option, so I'm not sure how the heck that's supposed to work...

2) And also, what if I'm intending on inserting multiple times in a loop? Is it going to create a new connection for every single INSERT_INTO operation?

Not knowing how to do this properly, since SQL.INSERT_INTO clearly can't just work out of the box, I went the route of the following, and so question  # 2 is something of importance:  

```
SqlBuilder builder=new SqlBuilder();
for(...) {
   builder.INSERT_INTO("my_table( my_column ) ")
   .VALUES("@colval",thevalue);
}
new Database(connString).Execute(builder);  // magic ?  who knows, it isn't documented...
```

Everything else looks great, just wish I had a bit more documentation. Can you confirm that the above is appropriate , or should I just trial and error this software..

**Comment by maxtoroq** [comment, marked as answer]
`SQL.INSERT_INTO()` is just a shortcut to `new SqlBuilder().INSERT_INTO()`.

About executing, you can choose different levels to work at:

**1. ADO.NET**: You can use [CreateCommand(DbConnection, SqlBuilder)](https://github.com/maxtoroq/DbExtensions/blob/v5/docs/api/DbExtensions/Extensions/CreateCommand_2.md) or [CreateCommand(DbProviderFactory, SqlBuilder)](https://github.com/maxtoroq/DbExtensions/blob/v5/docs/api/DbExtensions/Extensions/CreateCommand_5.md) to turn the SqlBuilder into a DbCommand. From that point on, you are responsible for opening the connection and calling ExecuteNonQuery.

**2. Database class**: The Database class makes things easier. You can call the Execute method like in your example. Database creates a connection when you pass a connection string in the constructor, or you can provide your own DbConnection. Database always uses the same [connection](https://github.com/maxtoroq/DbExtensions/blob/v5/docs/api/DbExtensions/Database/Connection.md), which opens and closes as necessary (you don't need to explicitly open the connection, but you are allowed to).

About executing multiple insert statements, calling Execute multiple times works, although there are two things you could do: 1. Use [EnsureConnectionOpen](https://github.com/maxtoroq/DbExtensions/blob/v5/docs/api/DbExtensions/Database/EnsureConnectionOpen.md), 2. Use TransactionScope.

**Comment by maxtoroq** [comment]
I just realized you are creating a batch insert, that's also possible, but you should append a semicolon at the end of each statement. e.g. `builder.INSERT_INTO().VALUES().Append(";")`

**Comment by ensemblebd** [comment]
awesome!!! thank you!! I saw the transaction system, and was wondering about that. Looks like the append is the way to go for batching them, that should do the trick. Have to say I am loving the flow, saves so much time. Now i just have to convert a couple hundred standard sql connections over to this system, and life will be good. 

I appreciate the fast support, and the assistance in thinking this through.  I should be on the right track now :)
